redaxo/src/core/lib/util/formatter.php

```diff
@@ -101,7 +101,7 @@ public static function strftime($value, $format = '')
         }
 
         // strftime does not exist anymore, return unformatted datetime string
-        if (is_int($value) || ctype_digit($value)) {
+        if (is_int($value) || preg_match('/^[0-9]+$/', $value) === 1) {
```

Comment:
Ich denke, das war vorher schon ok.
Mit String-Parametern arbeiten `ctype_digit` so, wie man es erwartet.
Nur wenn man ints übergibt, gibt es die überraschenden Effekte.

```
ctype_digit('15'); // true
ctype_digit('500'); // true

ctype_digit(15); // false -> hier die Überraschung
ctype_digit(500); // true
```

Wir übergeben aber gar keine ints, da wir vorher ja schon `is_int` abfragen.
Übergibt man einen Integer, erhält man seit 8.1 auch eine Deprecated-Meldung.

https://3v4l.org/QR843

Reply:
jep es ist kein aktues problem. wir hatten in einem anderen system probleme mit `ctype_digit`, weil kein `is_int` vorran gestellt war.

die idee des fixes ist, die einzigen stellen die das hier so verwendet zu entfernen damit niemand auf die idee kommt es woanders zu verwenden, weil er es hier gesehen hat

Reply:
Ich finde, wir nutzen die Methode genau so, wie sie gedacht ist. Zur Prüfung, ob ein String nur Ziffern enthält.
Finde da spricht nichts gegen. Und gerade durch die neue Deprecated-Meldung wird man ja auch auf die Fehlnutzung hingewiesen, wenn man Ints übergibt.

Reply:
wie gesagt: unsere nutzung hier ist valide.

man kann die methode aber sehr einfach falsch benutzen (hab ich heute schmerzlich erfahren)